br/pkg/storage/gcs.go

```diff
@@ -203,7 +201,7 @@ func (s *gcsStorage) WalkDir(ctx context.Context, opt *WalkOption, fn func(strin
 		// when walk on specify directory, the result include storage.Prefix,
 		// which can not be reuse in other API(Open/Read) directly.
 		// so we use TrimPrefix to filter Prefix for next Open/Read.
-		path := strings.TrimPrefix(attrs.Name, s.gcs.Prefix)
+		path := strings.TrimPrefix(attrs.Name, prefix)
```

Comment:
Could you add a test for it?

Reply:
I'm afraid `prefix` isn't equal to `s.gcs.Prefix` when `opt.SubDir` is set

Reply:
thanks you are right, I should not change the default behavior here, now this PR just adds more test cases for `WalkDir` ptal @3pointer

Reply:
Can we trim the extra slash prefix for the path?  It seems the behaviors of `gcs` and `s3` are not consistent with `local`.

Reply:
ping @gozssky  do you have more suggestions? if not I think this pr is ready to merge?

Reply:
LGTM. Please @3pointer take a look again.